### PR TITLE
feat(views): adds mini-framework for creating ajax-loaded popups with context

### PR DESCRIPTION
**Pull request by mrclay**
Plugins can now easily create ajax-loaded popups like Elgg’s hover menu, where the loaded view is rendered with the context, input, and page owner in which the trigger element was rendered.

Popups can be triggered by click or by hovering with a configurable delay.

BREAKING CHANGE:
This removes the `navigation/menu/user_hover/*` views

### CODE REVIEW

**Comment by mrclay** [comment]
Basically, all the popup/context data goes in the trigger, and the click/hover handler creates the $target on demand. The receiver view sets the old context and renders the view name based on type.

File: views/default/elements/navigation.css.php [hunk 1]
```diff
@@ -485,6 +485,9 @@
 	border-radius: 0 3px 3px 3px;
 	box-shadow: 1px 3px 5px rgba(0, 0, 0, 0.25);
 }
```
Comment (mrclay) [inline views/default/elements/navigation.css.php]: We just .elgg-menu-hover too generically. I'd rather put these props on .elgg-popup and define more specific selectors for our dropdown menus, user card, etc.

**Comment by hypeJunction** [comment]
Steve doesn't seem to have time. In case someone wants to salvage this: https://github.com/mrclay/Elgg-leaf/pull/8/commits/b6f36500011c242e2a987a12b052f267bd45eb10

**Comment by mrclay** [comment]
Just curious, was there no interest in this because bad timing?

**Comment by hypeJunction** [comment]
I think it was starting to blister our eyes in the list of PRa. I still think this should go in in some form

**Comment by jdalsem** [comment]
> because bad timing

yep... if it still is breaking it would need to wait until 3.1

**Pull request closed without merging**